### PR TITLE
Convert logs to use structured logging

### LINKED ISSUE #347: Add Support For Structured Logging
**Is your feature request related to a problem? Please describe.**

As a user of descheduler I'd like the descheduler to support generating structured logs, so that my log aggregation tools can more easily parse the descheduler logs. The k/k repo has recently started adding support for structured logs.

**Describe the solution you'd like**

Add a new `--logging-format` CLI option to descheduler to allow choosing between text and json log formats. The text log format should be the default.

Update klog usage in descheduler to use the new `InfoS` and `ErrorS` methods, https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#new-klog-methods.

**Describe alternatives you've considered**

Using the descheduler logs as is without any changes.

**What version of descheduler are you using?**

descheduler version: v0.18.0

**Additional context**

See structured logging KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging.

PR to add the `--logging-format` option to kube-scheduler: https://github.com/kubernetes/kubernetes/pull/91522.

### PR DESCRIPTION
**Pull request by farah**
Fixes #347

### CODE REVIEW

File: pkg/utils/predicates.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func podMatchesNodeLabels(pod *v1.Pod, node *v1.Node) bool {
 		// Match node selector for requiredDuringSchedulingIgnoredDuringExecution.
 		if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
 			nodeSelectorTerms := nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
-			klog.V(10).Infof("Match for RequiredDuringSchedulingIgnoredDuringExecution node selector terms %+v", nodeSelectorTerms)
+			klog.V(10).InfoS("Match for RequiredDuringSchedulingIgnoredDuringExecution node selector terms", "terms", nodeSelectorTerms)
```
Comment (seanmalloy) [inline pkg/utils/predicates.go:73]:
```suggestion
			klog.V(10).InfoS("Match for RequiredDuringSchedulingIgnoredDuringExecution node selector terms", "terms", klog.KObj(nodeSelectorTerms))
```

Can you verify if `klog.KObj()` can be used here?

**Review by seanmalloy — CHANGES_REQUESTED** [review]

File: pkg/utils/predicates.go (hunk 1 above)
Comment (farah) [reply pkg/utils/predicates.go:73]: @seanmalloy 

You cannot as you'll get a compiler error for missing methods `GetName` and `GetNamespace`.

```
func KObj(obj KMetadata) ObjectRef {
	return ObjectRef{
		Name:      obj.GetName(),
		Namespace: obj.GetNamespace(),
	}
}
```

**Comment by ingvagabund** [comment]
LGTM so far

File: pkg/descheduler/evictions/evictions.go [hunk 2]
```diff
@@ -112,9 +112,9 @@ func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node,
 
 	pe.nodepodCount[node]++
 	if pe.dryRun {
-		klog.V(1).Infof("Evicted pod in dry run mode: %#v in namespace %#v%s", pod.Name, pod.Namespace, reason)
+		klog.V(1).InfoS("Evicted pod in dry run mode", "pod", klog.KObj(pod), "namespace", pod.Namespace, "reason", reason)
```
Comment (seanmalloy) [inline pkg/descheduler/evictions/evictions.go:115]:
```suggestion
		klog.V(1).InfoS("Evicted pod in dry run mode", "pod", klog.KObj(pod), "reason", reason)
```

File: pkg/descheduler/evictions/evictions.go [hunk 3]
```diff
@@ -112,9 +112,9 @@ func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node,
...
 	} else {
-		klog.V(1).Infof("Evicted pod: %#v in namespace %#v%s", pod.Name, pod.Namespace, reason)
+		klog.V(1).InfoS("Evicted pod", "pod", klog.KObj(pod), "namespace", pod.Namespace, "reason", reason)
```
Comment (seanmalloy) [inline pkg/descheduler/evictions/evictions.go:117]:
```suggestion
		klog.V(1).InfoS("Evicted pod", "pod", klog.KObj(pod), "reason", reason)
```

**Comment by seanmalloy** [comment]
@farah I added two more small suggestions. After these are fixed I'll add my LGTM. Thanks!

File: pkg/descheduler/evictions/evictions.go (hunk 3 above)
Comment (farah) [reply pkg/descheduler/evictions/evictions.go]: done

File: pkg/descheduler/evictions/evictions.go (hunk 2 above)
Comment (farah) [reply pkg/descheduler/evictions/evictions.go]: done

**Review by seanmalloy — APPROVED** [review]

**Review by damemi — APPROVED** [review]
/approve

**Pull request merged**